src/Messages/OWSUnknownContactBlockOfferMessage.m

```diff
@@ -34,14 +34,16 @@ - (instancetype)initWithTimestamp:(uint64_t)timestamp thread:(TSThread *)thread
     return self;
 }
 
-- (nullable NSDate *)receiptDateForSorting
+- (BOOL)shouldUseReceiptDateForSorting
 {
-    // Always use date, since we're creating these interactions after the fact
-    // and back-dating them.
-    //
-    // By default [TSMessage receiptDateForSorting] will prefer to use receivedAtDate
-    // which is not back-dated.
-    return self.date;
+    // Use the timestamp, not the "received at" timestamp to sort,
+    // since we're creating these interactions after the fact and back-dating them.
+    return NO;
+}
+
+- (BOOL)isDynamicInteraction
```

Comment:
I thought about it. It's true today, but it seemed like the kind of thing that could introduce a very-hard to debug issue if either of us forgot or moved on.